src/libraries/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -115,7 +115,12 @@ private static string MarshalToString(IntPtr arg, string argName)
                                                     IntPtr functionHandle)
         {
             if (!IsSupported)
+            {
+#if CORECLR
+                OnDisabledGetFunctionPointerCall(typeNameNative, methodNameNative);
```

Comment:
This can throw if one of the names is null. We should never leak a managed exception to native code, unless we're sure it's OK. It is OK in the CPP/CLI case, but it's NOT OK for normal native hosts.
So this will need a try/catch and return HRESULT around it unfortunately.
I would do it here (instead of inside the callback) - as the contract about no exceptions is this method's responsibility.

Reply:
You're right - I missed that. But I still think having the try/catch is a good idea :wink: